error while compiling

hello i have a few errors while compiling on platformIO, 
the file I opened with the "open project" button is the pure one, with no modifications that I downloaded from here. the errors are the following:

In file included from Marlin\src\HAL\STM32F1\dogm\../../../inc/../HAL/./STM32F1/HAL.h:36,
                 from Marlin\src\HAL\STM32F1\dogm\../../../inc/../HAL/HAL.h:26,
                 from Marlin\src\HAL\STM32F1\dogm\../../../inc/MarlinConfig.h:30,
                 from Marlin\src\HAL\STM32F1\dogm\u8g_com_stm32duino_fsmc.cpp:29:
Marlin\src\HAL\STM32F1\dogm\../../../inc/../HAL/./STM32F1/fastio.h:30:10: fatal error: libmaple/gpio.h: No such file or directory
   30 | #include <libmaple/gpio.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio\build\mks_robin_nano35\src\src\HAL\STM32F1\dogm\u8g_com_stm32duino_fsmc.cpp.o] Error 1
In file included from Marlin\src\HAL\STM32F1\../../inc/../HAL/./STM32F1/HAL.h:36,
                 from Marlin\src\HAL\STM32F1\../../inc/../HAL/HAL.h:26,
                 from Marlin\src\HAL\STM32F1\../../inc/MarlinConfig.h:30,
                 from Marlin\src\HAL\STM32F1\onboard_sd.cpp:15:
Marlin\src\HAL\STM32F1\../../inc/../HAL/./STM32F1/fastio.h:30:10: fatal error: libmaple/gpio.h: No such file or directory
   30 | #include <libmaple/gpio.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio\build\mks_robin_nano35\src\src\HAL\STM32F1\onboard_sd.cpp.o] Error 1
In file included from Marlin\src\HAL\shared\../../inc/../HAL/./STM32F1/HAL.h:36,
                 from Marlin\src\HAL\shared\../../inc/../HAL/HAL.h:26,
                 from Marlin\src\HAL\shared\../../inc/MarlinConfig.h:30,
                 from Marlin\src\HAL\shared\HAL_spi_L6470.cpp:28:
Marlin\src\HAL\shared\../../inc/../HAL/./STM32F1/fastio.h:30:10: fatal error: libmaple/gpio.h: No such file or directory
   30 | #include <libmaple/gpio.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio\build\mks_robin_nano35\src\src\HAL\shared\HAL_spi_L6470.cpp.o] Error 1
In file included from Marlin\src\HAL\shared\backtrace\../../../core/../inc/../HAL/./STM32F1/HAL.h:36,
                 from Marlin\src\HAL\shared\backtrace\../../../core/../inc/../HAL/HAL.h:26,
                 from Marlin\src\HAL\shared\backtrace\../../../core/../inc/MarlinConfig.h:30,
                 from Marlin\src\HAL\shared\backtrace\../../../core/serial.h:24,
                 from Marlin\src\HAL\shared\backtrace\backtrace.cpp:29:
Marlin\src\HAL\shared\backtrace\../../../core/../inc/../HAL/./STM32F1/fastio.h:30:10: fatal error: libmaple/gpio.h: No such file or directory
   30 | #include <libmaple/gpio.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio\build\mks_robin_nano35\src\src\HAL\shared\backtrace\backtrace.cpp.o] Error 1
In file included from Marlin\src\HAL\shared\backtrace\../../../inc/../HAL/./STM32F1/HAL.h:36,
                 from Marlin\src\HAL\shared\backtrace\../../../inc/../HAL/HAL.h:26,
                 from Marlin\src\HAL\shared\backtrace\../../../inc/MarlinConfig.h:30,
                 from Marlin\src\HAL\shared\backtrace\unwmemaccess.cpp:18:
Marlin\src\HAL\shared\backtrace\../../../inc/../HAL/./STM32F1/fastio.h:30:10: fatal error: libmaple/gpio.h: No such file or directory     
   30 | #include <libmaple/gpio.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio\build\mks_robin_nano35\src\src\HAL\shared\backtrace\unwmemaccess.cpp.o] Error 1
====================================================== [FAILED] Took 47.98 seconds ======================================================

Environment               Status    Duration
------------------------  --------  ------------
mega2560                  IGNORED
mega1280                  IGNORED
rambo                     IGNORED
FYSETC_F6_13              IGNORED
FYSETC_F6_14              IGNORED
sanguino644p              IGNORED
sanguino1284p             IGNORED
melzi                     IGNORED
melzi_optiboot            IGNORED
at90usb1286_cdc           IGNORED
at90usb1286_dfu           IGNORED
DUE                       IGNORED
DUE_USB                   IGNORED
DUE_debug                 IGNORED
LPC1768                   IGNORED
LPC1769                   IGNORED
STM32F103RC               IGNORED
STM32F103RC_fysetc        IGNORED
STM32F103RC_btt           IGNORED
STM32F103RC_btt_USB       IGNORED
STM32F103RC_btt_512K      IGNORED
STM32F103RC_btt_512K_USB  IGNORED
STM32F103RE               IGNORED
STM32F103RE_btt           IGNORED
STM32F103RE_btt_USB       IGNORED
STM32F4                   IGNORED
STM32F7                   IGNORED
ARMED                     IGNORED
STM32F103VE_GTM32         IGNORED
STM32F103VE_longer        IGNORED
mks_robin_mini            IGNORED
mks_robin_nano35          FAILED    00:00:47.976
mks_robin                 IGNORED
mks_robin_pro35           IGNORED
mks_robin_lite            IGNORED
mks_robin_lite3           IGNORED
jgaurora_a5s_a1           IGNORED
STM32F103CB_malyan        IGNORED
chitu_f103                IGNORED
STM32F401VE_STEVAL        IGNORED
FLYF407ZG                 IGNORED
FYSETC_S6                 IGNORED
STM32F407VE_black         IGNORED
BIGTREE_SKR_PRO           IGNORED
BIGTREE_GTR_V1_0          IGNORED
BIGTREE_BTT002            IGNORED
teensy31                  IGNORED
teensy35                  IGNORED
esp32                     IGNORED
linux_native              IGNORED
SAMD51_grandcentral_m4    IGNORED
rumba32_f446ve            IGNORED
rumba32_mks               IGNORED
include_tree              IGNORED
================================================= 1 failed, 0 succeeded in 00:00:47.976 =================================================
The terminal process terminated with exit code: 1

--- a/platformio.ini
+++ b/platformio.ini
@@ -244,6 +244,44 @@ lib_deps          = Servo
   TMCStepper@>=0.6.1,<1.0.0
   Adafruit NeoPixel=https://github.com/p3p/Adafruit_NeoPixel/archive/release.zip
   SailfishLCD=https://github.com/mikeshub/SailfishLCD/archive/master.zip
+  
+  
+#################################
+#                               #
+#      STM32 Architecture       #
+#                               #
+#################################
+#
+# HAL/STM32 Base Environment values
+#
+[common_stm32]
+platform      = ststm32@<6.2.0
+platform_packages = framework-arduinoststm32@>=4.10700,<4.10800
+lib_deps      = ${common.lib_deps}
+lib_ignore    = SoftwareSerial
+build_flags   = ${common.build_flags}
+  -IMarlin/src/HAL/STM32 -std=gnu++14
+  -DUSBCON -DUSBD_USE_CDC
+  -DUSBD_VID=0x0483
+  -DTIM_IRQ_PRIO=13
+build_unflags = -std=gnu++11
+src_filter    = ${common.default_src_filter} +<src/HAL/STM32>
+#
+# HAL/STM32F1 Common Environment values
+#
+[common_stm32f1]
+platform      = ${common_stm32.platform}
+build_flags   = !python Marlin/src/HAL/STM32F1/build_flags.py
+  ${common.build_flags} -std=gnu++14 -DHAVE_SW_SERIAL
+build_unflags = -std=gnu++11
+src_filter    = ${common.default_src_filter} +<src/HAL/STM32F1>
+lib_ignore    =
+  Adafruit NeoPixel
+  SPI
+lib_deps      = ${common.lib_deps}
+  SoftwareSerialM
+
+
 
 #
 # STM32F103RC
@@ -480,21 +518,26 @@ src_filter    = ${common.default_src_filter} +<src/HAL/STM32F1>
 lib_ignore    = Adafruit NeoPixel, SPI
 
 #
-# MKS Robin Nano (STM32F103VET6)
+# MKS Robin Nano (STM32F103VET6) - Emulated Graphical 128x64 (DOGM) UI
 #
-[env:mks_robin_nano35]
-platform      = ststm32
+[env:mks_robin_nano]
+platform      = ${common_stm32f1.platform}
+extends       = common_stm32f1
 board         = genericSTM32F103VE
 platform_packages = tool-stm32duino
-build_flags   = !python Marlin/src/HAL/STM32F1/build_flags.py
-  ${common.build_flags} -std=gnu++14 -DHAVE_SW_SERIAL -DSS_TIMER=4
-build_unflags = -std=gnu++11
+extra_scripts = buildroot/share/PlatformIO/scripts/mks_robin_nano.py
+build_flags   = ${common_stm32f1.build_flags}
+  -DMCU_STM32F103VE -DSS_TIMER=4
+
+#
+# MKS Robin Nano (STM32F103VET6) - MKS UI (LVGL)
+#
+[env:mks_robin_nano35]
+platform      = ${common_stm32f1.platform}
+extends       = env:mks_robin_nano
 extra_scripts = buildroot/share/PlatformIO/scripts/mks_robin_nano35.py
-src_filter    = ${common.default_src_filter} +<src/HAL/STM32F1>
-lib_deps      = ${common.lib_deps}
-  SoftwareSerialM=https://github.com/FYSETC/SoftwareSerialM/archive/master.zip
+lib_deps      = ${common_stm32f1.lib_deps}
   MKS-LittlevGL=https://github.com/makerbase-mks/MKS-LittlevGL/archive/master.zip
-lib_ignore    = Adafruit NeoPixel, SPI
 debug_tool = jlink
 upload_protocol = jlink
 
@@ -502,18 +545,11 @@ upload_protocol = jlink
 # MKS Robin Nano v2.0 (STM32F103VET6)
 #
 [env:mks_robin_nano35_v2]
-platform      = ststm32
-board         = genericSTM32F103VE
-platform_packages = tool-stm32duino
-build_flags   = !python Marlin/src/HAL/STM32F1/build_flags.py
-  ${common.build_flags} -std=gnu++14 -DHAVE_SW_SERIAL -DSS_TIMER=4
-build_unflags = -std=gnu++11
+platform      = ${common_stm32f1.platform}
+extends       = env:mks_robin_nano
 extra_scripts = buildroot/share/PlatformIO/scripts/mks_robin_nano35_v2.py
-src_filter    = ${common.default_src_filter} +<src/HAL/STM32F1>
-lib_deps      = ${common.lib_deps}
-  SoftwareSerialM=https://github.com/FYSETC/SoftwareSerialM/archive/master.zip
+lib_deps      = ${common_stm32f1.lib_deps}
   MKS-LittlevGL=https://github.com/makerbase-mks/MKS-LittlevGL/archive/master.zip
-lib_ignore    = Adafruit NeoPixel, SPI
 debug_tool = jlink
 upload_protocol = jlink
 

